hazelcast/src/test/java/com/hazelcast/map/EntryProcessorOffloadableTest.java

```diff
@@ -928,6 +938,110 @@ public String getExecutorName() {
         }
     }
 
+    /**
+     * <pre>
+     * Given: Heart beats are configured to come each few seconds (i.e. one quarter of hazelcast.operation.call.timeout.millis
+     *        - set in the {@code getConfig()} method)
+     * When: An offloaded EntryProcessor takes a long time to run.
+     * Then: Heart beats are still coming during the task is offloaded.
+     * </pre>
+     *
+     * @see #getConfig()
+     * @see #HEARTBEATS_INTERVAL_SEC
+     */
+    @Test
+    public void testHeartBeatsCommingWhenEntryPropcessorOffloaded() throws InterruptedException, IOException {
+        final String key = generateKeyOwnedBy(instances[1]);
+        SimpleValue givenValue = new SimpleValue(1);
+
+        final IMap<Object, Object> map = instances[0].getMap(MAP_NAME);
+        map.put(key, givenValue);
+
+        final Address instance1Address = instances[1].getCluster().getLocalMember().getAddress();
+        final AtomicInteger heartBeatsCounter = new AtomicInteger(0);
+        final AtomicLong minDelta = new AtomicLong(Long.MAX_VALUE);
+        Thread hbMonitorThread = new Thread() {
+            public void run() {
+                OperationServiceImpl osImpl = (OperationServiceImpl) HazelcastTestSupport.getNodeEngineImpl(instances[0])
+                        .getOperationService();
+                Map<Address, AtomicLong> heartBeats = osImpl.getInvocationMonitor().getHeartbeatPerMember();
+                long lastbeat = Long.MIN_VALUE;
+                try {
+                    while (!isInterrupted()) {
+                        AtomicLong timestamp = heartBeats.get(instance1Address);
+                        if (timestamp != null) {
+                            long newlastbeat = timestamp.get();
+                            if (lastbeat != newlastbeat) {
+                                long delta = newlastbeat - lastbeat;
+                                if (lastbeat != Long.MIN_VALUE && delta < minDelta.get()) {
+                                    minDelta.set(delta);
+                                }
+                                lastbeat = newlastbeat;
+                                heartBeatsCounter.incrementAndGet();
+                            }
+                        }
+                        Thread.sleep(100L);
```

Comment:
You should be able to use `HazelcastTestSupport.sleepMillis()` here, which restores the interrupted flag. So you can get rid of the `try-catch` block and still monitor `isInterrupted()`, if I'm not wrong.